### PR TITLE
Added task controller.

### PR DESCRIPTION
**Pull request by makaroshyna**

### CODE REVIEW

File: src/main/java/app/controller/TaskController.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Task management", description = "Endpoints for managing tasks")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/projects/{projectId}/tasks")
```
Comment (VlasenkoYevhenii) [inline src/main/java/app/controller/TaskController.java:26]: image

Comment (VlasenkoYevhenii) [reply src/main/java/app/controller/TaskController.java:26]: fix paths and related methods if needed

File: src/main/java/app/dto/task/CreateTaskRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import java.time.LocalDate;
+import lombok.Data;
+import org.springframework.format.annotation.DateTimeFormat;
+
+@Data
+public class CreateTaskRequestDto {
+    @NotBlank
+    private String name;
```
Comment (VlasenkoYevhenii) [inline src/main/java/app/dto/task/CreateTaskRequestDto.java:15]:
```suggestion
    @NotBlank(message = "Task name must be defined"
    private String name;
```
also check other fields

File: src/main/java/app/service/task/TaskServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    private final ProjectRepository projectRepository;
+    private final CommentService commentService;
+
+    @Override
+    public TaskDtoWithoutLabelsAndComments save(CreateTaskRequestDto requestDto, Long projectId) {
+        Project project = getProjectById(projectId);
+        Task task = taskMapper.toModel(requestDto);
+        task.setProject(project);
+        task.setComments(Set.of());
+        task.setLabels(List.of());
```
Comment (VlasenkoYevhenii) [inline src/main/java/app/service/task/TaskServiceImpl.java:32]: extract this logic into a separate method (optional)

**Review by VlasenkoYevhenii — CHANGES_REQUESTED** [review]
GJ!
some fixes still required

File: src/main/java/app/controller/TaskController.java (hunk 1 above)
Comment (makaroshyna) [reply src/main/java/app/controller/TaskController.java:26]: But how would I get tasks for a project if there if no project ID in the path?

Comment (makaroshyna) [reply src/main/java/app/controller/TaskController.java:26]: I asked in chat earlier and me and Valik came to the conclusion that it was better to do it like that. If needed I will change this path later

Comment (VlasenkoYevhenii) [reply src/main/java/app/controller/TaskController.java:26]: `/api/tasks/byproject/{projectId}` ? or something like that
with this path `/api/projects/{projectId}/tasks` you're requesting something from task-related logic but in fact, it's the path for `ProjectController`.
You will get the same path variable but the path hierarchy will be consistent

**Review by VlasenkoYevhenii — CHANGES_REQUESTED** [review]

**Review by VlasenkoYevhenii — APPROVED** [review]

**Review by Bohadan-Markatov — APPROVED** [review]
lgtm

**Pull request merged**